EventsExpress.Core/Services/TokenService.cs

```diff
@@ -0,0 +1,153 @@
+            _signingEncodingKey = jwtSigningEncodingKey;
+            _userService = userService;
+            _httpContextAccessor = httpContextAccessor;
+            _mapper = mapper;
+        }
+        public string GenerateAccessToken(UserDTO user)
+        {
+            var lifeTime = _configuration.GetValue<int>("JWTOptions:LifeTime");
+            var claims = new[]
+            {
+                new Claim(ClaimTypes.Email, user.Email),
+                new Claim(ClaimTypes.Role, user.Role.Name),
+                new Claim(ClaimTypes.Name, user.Id.ToString()),
+            };
+            var jwtToken = new JwtSecurityToken(
+                claims: claims,
+                expires: DateTime.Now.AddSeconds(lifeTime),
+                signingCredentials: new SigningCredentials(
+                        _signingEncodingKey.GetKey(),
+                        _signingEncodingKey.SigningAlgorithm)
+            );
+            return new JwtSecurityTokenHandler().WriteToken(jwtToken);
+        }
+        public RefreshToken GenerateRefreshToken(string ipAddress)
+        {
+            var randomNumber = new byte[32];
+            using (var rng = RandomNumberGenerator.Create())
+            {
+                rng.GetBytes(randomNumber);
+                return new RefreshToken
+                {
+                    Token = Convert.ToBase64String(randomNumber),
+                    Expires = DateTime.Now.AddDays(7),
+                    Created = DateTime.Now,
+                    CreatedByIp = ipAddress
+                };
+            }
+        }
+        public ClaimsPrincipal GetPrincipalFromExpiredToken(string token)
+        {
+            var signingKey = new SigningSymmetricKey(_configuration.GetValue<string>("JWTOptions:SecretKey"));
+
+            var signingDecodingKey = (IJwtSigningDecodingKey)signingKey;
+
+            var tokenValidationParameters = new TokenValidationParameters
+            {
+                ValidateAudience = false,
+                ValidateIssuer = false,
+                ValidateIssuerSigningKey = true,
+                IssuerSigningKey = signingDecodingKey.GetKey(),
+                ValidateLifetime = false //here we are saying that we don't care about the token's expiration date
+            };
+            var tokenHandler = new JwtSecurityTokenHandler();
+            var principal = tokenHandler.ValidateToken(token, tokenValidationParameters, out SecurityToken securityToken);
+            if (!(securityToken is JwtSecurityToken jwtSecurityToken) || !jwtSecurityToken.Header.Alg.Equals(SecurityAlgorithms.HmacSha256, StringComparison.InvariantCultureIgnoreCase))
+                throw new SecurityTokenException("Invalid token");
+            return principal;
+        }
+        public async Task<AuthenticateResponseModel> RefreshToken(string token)
+        {
+            var user = _userService.GetUserByRefreshToken(token);
+            // return null if no user found with token
```

Comment:
I think it isn't good to return NULL here. We must be able to say more apparently : "Failed".
Consider possibility to inherit OperationdDetails